packages/sitecore-jss-angular/src/components/base-field.directive.ts

```diff
@@ -0,0 +1,32 @@
+import { Directive, Type, ViewContainerRef, EmbeddedViewRef, TemplateRef } from '@angular/core';
+import { RenderingField } from './rendering-field';
+import { GenericFieldValue, isFieldValueEmpty } from '@sitecore-jss/sitecore-jss/layout';
+
+@Directive()
+export abstract class BaseFieldDirective {
+  protected viewRef: EmbeddedViewRef<unknown>;
+  protected abstract field: RenderingField<GenericFieldValue>;
+  protected abstract emptyFieldEditingTemplate: TemplateRef<unknown>;
```

Comment:
Can we just define an "input" here?
```
@Input('scDateEmptyFieldEditingTemplate') emptyFieldEditingTemplate: TemplateRef<unknown>;
```
so, we don't need to define this property on field level, since it's not used directly by field

Reply:
Also, is it possible to initialize this input with the default value passed by the inherited class via the constructor? So we don't need to pass the default component via "renderEmpty" function

Reply:
On your second comment: makes sense, Updated.
On your first comment: I tried but seems structural directives require the names of their inputs to be prefixed by the directive selector (see how the property is common but the input alias is different for each directive) - and I could not find a way around that; you can see it is done the same way for the existing inputs in [link.directive](https://github.com/Sitecore/jss/blob/42c071d29fdc35ce2db223b5afe368fab986b0ae/packages/sitecore-jss-angular/src/components/link.directive.ts) and [generic link directive](https://github.com/Sitecore/jss/blob/42c071d29fdc35ce2db223b5afe368fab986b0ae/packages/sitecore-jss-angular/src/components/link.directive.ts)